src/scalar.rs

```diff
@@ -624,9 +624,12 @@ impl Scalar {
         res
     }
 
-    /// Computes the multiplicative inverse of this element,
-    /// failing if the element is zero.
+    /// Computes the multiplicative inverse of this element, failing if the
+    /// element is zero.
     pub fn invert(&self) -> CtOption<Self> {
+        if *self == Scalar::one() || *self == -Scalar::one() {
+            return CtOption::new(*self, !self.ct_eq(&Self::zero()));
```

Comment:
Why do we check if `self.ct_eq(&Self::zero())`, considering we already know the scalar is either `-1` or `1`

Reply:
For no reason, will change it to `true`